### DISCUSSION TITLE
Unit 2 Project: Angle of Incidence equals Angle of Reflection

### DISCUSSION THREAD

**Discussion by degriff** [TEALSK12/introduction-to-computer-science#343] [Ideas]
This is my third year teaching this course.  My observation is that students tend to struggle with the math behind the paddle strike.  I believe that part of the problem is the Do Now introduces the `if on edge, bounce` block.  While this block is helpful for the behavior for upper/lower edges of the stage, it is a source of frustration for the left/right edges and paddles.  I suggest giving them this code snippet to help the majority of students with the ball touching a paddle.

![paddle reflection](https://user-images.githubusercontent.com/7989039/136968341-31a98203-4d66-4c03-afab-0cc613882e0f.png)

**Comment by jdonwells** [comment]
Why not simplify to 0 - direction? The math behind that is simpler to explain. You turn the same amount in the opposite direction.

(270 + (90 - direction)) modulo 360
= (360 - direction) modulo 360
= - direction

To avoid the problems with students being unprepared for Pong I took 15 minutes in each of 6 consecutive workdays to build a Breakout game. I noticed students using the concepts introduced in Breakout in Pong after they are shown. Breakout is close enough to be helpful with Pong but different enough to require understanding to apply the concepts.

**Reply by degriff** [reply]
I like that solution better.

I think it does obfuscates the underlying issue that most of the students that I have worked with in this program struggle with geometry.  They needed more guidance than what is provided in the lesson to draw the shapes in Lab 1.3.  Identifying the different types of triangles in Lab 2.5 is not sufficient to work out the reflection angle when striking the paddle.  The goal of this project is to give the opportunity to practice the skills and concepts from this unit: Variables and Operators.  Anything else is noise.

For some additional background, this curriculum is based on 55 minute class period.  At my first school, we had 49 minute periods with a 36 minute period once a month.  My current school has 45 minute periods with a 42 minute period once a week.  It is a challenge to maintain a pace to provide the information and to give them enough time for the formative project work.  I believe the teachers we are attempting to train need the above code snippet and slides to help work through the math so they can choose the best option for their students.

**Reply by karlmath10** [reply]
Would you be willing to share your code for the breakout game? I am thinking of doing the same. What I am most interested in is getting the ball to bounce off different surfaces of the bricks in different ways. This is troublesome with the paddles in pong as well.

**Comment by jdonwells** [comment]
These are the steps to build Breakout live in front of the class.  Keep in mind there is an old saying among people who do presentations at CS conferences _"It takes a special kind of arrogance to code in front of an audience"_.

https://snap.berkeley.edu/snap/snap.html#present:Username=jdonwells&ProjectName=Breakout%20Game%20Starter%20Code%20A

https://snap.berkeley.edu/snap/snap.html#present:Username=jdonwells&ProjectName=Breakout%20Game%20Starter%20Code%20B

https://snap.berkeley.edu/snap/snap.html#present:Username=jdonwells&ProjectName=Breakout%20Game%20Starter%20Code%20C

https://snap.berkeley.edu/snap/snap.html#present:Username=jdonwells&ProjectName=Breakout%20Game%20Starter%20Code%20D

https://snap.berkeley.edu/snap/snap.html#present:Username=jdonwells&ProjectName=Breakout%20Game%20Starter%20Code%20E

https://snap.berkeley.edu/snap/snap.html#present:Username=jdonwells&ProjectName=Breakout%20Game%20Starter%20Code%20F

https://snap.berkeley.edu/snap/snap.html#present:Username=jdonwells&ProjectName=Breakout%20Demo%20Easy%20Final%20Game%20

**Comment by karlmath10** [comment]
Thank you very much! I really appreciate your help!

On Thu, Sep 14, 2023 at 9:59 AM Don Wells ***@***.***> wrote:

> These are the steps to build Breakout live in front of the class. Remember
> there is an old saying among Computer Science teachers *"It takes a
> special kind of arrogance to code in front of an audience"*.
>
>
> https://snap.berkeley.edu/snap/snap.html#present:Username=jdonwells&ProjectName=Breakout%20Game%20Starter%20Code%20A
>
>
> https://snap.berkeley.edu/snap/snap.html#present:Username=jdonwells&ProjectName=Breakout%20Game%20Starter%20Code%20B
>
>
> https://snap.berkeley.edu/snap/snap.html#present:Username=jdonwells&ProjectName=Breakout%20Game%20Starter%20Code%20C
>
>
> https://snap.berkeley.edu/snap/snap.html#present:Username=jdonwells&ProjectName=Breakout%20Game%20Starter%20Code%20D
>
>
> https://snap.berkeley.edu/snap/snap.html#present:Username=jdonwells&ProjectName=Breakout%20Game%20Starter%20Code%20E
>
>
> https://snap.berkeley.edu/snap/snap.html#present:Username=jdonwells&ProjectName=Breakout%20Game%20Starter%20Code%20F
>
>
> https://snap.berkeley.edu/snap/snap.html#present:Username=jdonwells&ProjectName=Breakout%20Demo%20Easy%20Final%20Game%20
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/TEALSK12/introduction-to-computer-science/discussions/343#discussioncomment-7002169>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJY7553YH756Z2Y6JDZFWDTX2METTANCNFSM5F2UDRLA>
> .
> You are receiving this because you commented.Message ID:
> <TEALSK12/introduction-to-computer-science/repo-discussions/343/comments/7002169
> @github.com>
>


-- 
*Daniel Karl*
Computer Science and Mathematics Teacher
(740) 852-5705 ext. 3039

*London High School*
336 Elm Street
London, OH 43140
https://www.london.k12.oh.us/

**Comment by jdonwells** [comment]
There are presentations to go with the code. 
https://docs.google.com/presentation/d/1maPcD3naM-R9zT1O9_qy1tCApIV2AeCElearPA117L4/edit?usp=sharing
https://docs.google.com/presentation/d/1TpIjX7WhymV8RonqDEkBDVZ7QM0SbrIVZqJnzPZARNw/edit?usp=sharing
https://docs.google.com/presentation/d/1k3W9h_dNnEmeK9b6ZAaUkgTyNvqN2oysjJXx7Ro5tkw/edit?usp=sharing
https://docs.google.com/presentation/d/1fEsvhYI04blZ_wqUk6iqDN55T_-YQE8uy_-pqiVeLdo/edit?usp=sharing
https://docs.google.com/presentation/d/1aFMOfSNqzi_lH2nE7o3Sz6SdtnfEWKOs_8Lp3CYljLY/edit?usp=sharing
https://docs.google.com/presentation/d/1FBlWNYupYg04a3jsyKxuX42WzKmuO7DEutcWW6xOYYU/edit?usp=sharing

Lectures B and C code up the bounce off the paddle. Lecture C I use clones to create the bricks. They don't know that yet, but I do it anyway.
